### PR TITLE
Adds the ability to make fingerless gloves

### PR DESCRIPTION
**Pull request by Carbadox**
## About the PR 
Adds the ability to make fingerless gloves by using snipping tools on black gloves
Very important feature

## Why's this needed? 
sandbox, they use nearly the same sprite, it'd be expected to be able to make fingerless gloves

## Changelog

```changelog
(u)Carbadox
(+)Added the ability to make fingerless gloves by using snipping tools on black gloves.
```

### CODE REVIEW

File: code/obj/item/clothing/gloves.dm [hunk 1]
```diff
@@ -234,6 +234,16 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
 		..()
 		setProperty("heatprot", 7)
 
+/obj/item/clothing/gloves/black/attackby(obj/item/W as obj, mob/user as mob)
+	if (istool(W, TOOL_CUTTING | TOOL_SNIPPING))
+		user.visible_message("<span class='notice'>[user] cuts off the fingertips from the [src].</span>")
```
Comment (flrsh) [inline code/obj/item/clothing/gloves.dm:239]:
```suggestion
		user.visible_message("<span class='notice'>[user] cuts off the fingertips from [src].</span>")
```

File: code/obj/item/clothing/gloves.dm [hunk 2]
```diff
@@ -234,6 +234,16 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
...
+		if(src.loc == user)
+			user.u_equip(src)
+		src.set_loc(user)
+		var/obj/item/clothing/gloves/black/I = new /obj/item/clothing/gloves/fingerless
+		user.put_in_hand_or_drop(I)
+		qdel(src)
+
```
Comment (flrsh) [inline code/obj/item/clothing/gloves.dm:246]:
```suggestion
	else . = ..()
```

**Review by flrsh — CHANGES_REQUESTED** [review]

**Comment by Kyle2143** [comment]
Yes, good idea. But do flourish's changes first yeah.

File: code/obj/item/clothing/gloves.dm [hunk 3]
```diff
@@ -234,6 +234,16 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
...
 
+/obj/item/clothing/gloves/black/attackby(obj/item/W as obj, mob/user as mob)
+	if (istool(W, TOOL_CUTTING | TOOL_SNIPPING))
+		user.visible_message("<span class='notice'>[user] cuts off the fingertips from the [src].</span>")
+		if(src.loc == user)
+			user.u_equip(src)
+		src.set_loc(user)
+		var/obj/item/clothing/gloves/black/I = new /obj/item/clothing/gloves/fingerless
+		user.put_in_hand_or_drop(I)
+		qdel(src)
```
Comment (Sovexe) [inline code/obj/item/clothing/gloves.dm:245]: * your type casting this object to black gloves needlessly. That said, you don't actually even need that var
* setting the black gloves `loc` to the user contents is also unneeded as `qdel(src)` already warps the atom to null space

```suggestion
		qdel(src)
		user.put_in_hand_or_drop(new /obj/item/clothing/gloves/fingerless)
```

File: code/obj/item/clothing/gloves.dm [hunk 4]
```diff
@@ -234,6 +234,16 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
 		..()
 		setProperty("heatprot", 7)
 
+/obj/item/clothing/gloves/black/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (Sovexe) [inline code/obj/item/clothing/gloves.dm:237]:
```suggestion
/obj/item/clothing/gloves/black/attackby(obj/item/W, mob/user)
```
Redundant and might even hurt performance (for however many fractions of a micro second)

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/obj/item/clothing/gloves.dm (hunk 3 above)
Comment (Carbadox) [reply code/obj/item/clothing/gloves.dm:245]: is the double qdel on purpose?

Comment (Sovexe) [reply code/obj/item/clothing/gloves.dm:245]: I'm unsure where there is a second. I removed one and moved it up.

**Review by flrsh — APPROVED** [review]

**Review by Sovexe — APPROVED** [review]

**Pull request merged**